xonsh/tools.py

```diff
@@ -346,6 +348,44 @@ def subproc_toks(line, mincol=-1, maxcol=None, lexer=None, returnline=False):
     return rtn
 
 
+def check_bad_str_token(tok):
+    """Checks if a token is a bad string."""
+    if tok.type == 'ERRORTOKEN' and tok.value == 'EOF in multi-line string':
+        return True
+    elif isinstance(tok.value, str) and not check_quotes(tok.value):
+        return True
+    else:
+        return False
+
+
+def check_quotes(s):
+    """Checks a string to make sure that if it starts with quotes, it also
+    ends with quotes.
+    """
+    starts_as_str = RE_BEGIN_STRING.match(s) is not None
+    ends_as_str = s.endswith('"') or s.endswith("'")
+    if not starts_as_str and not ends_as_str:
+        ok = True
+    elif starts_as_str and not ends_as_str:
+        ok = False
+    elif not starts_as_str and ends_as_str:
+        ok = False
+    else:
+        m = RE_COMPLETE_STRING.match(s)
+        ok = m is not None
+    return ok
+
+
+def _have_open_triple_quotes(s):
+    if s.count('"""') % 2 == 1:
+        open_triple = '"""'
+    elif s.count("'''") % 2 == 1:
+        open_triple = "'''"
```

Comment:
So the `''' I like """ quotes '''` case should not come up because that is a valid Python string. `get_logical_lines()` is only called when there is a SyntaxError and so it won't trigger. This really only comes up when someone has typed in a partial string, like `''' I like """ qu`, and then needs a continuation line. 

In the worst case, `get_logical_lines()` will treat the rest of the input as a single line, eating up the rest of the input the user wrote at the command line. However, `subroc_toks()` then re-tokeninzes the line and will correctly see the complete string and newlines.